File: src/main/java/com/example/demo/model/Product.java
package com.example.demo.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.util.Collection;
@Getter
@Setter
@EqualsAndHashCode
@ToString
@Entity
@Table(name="Product")
public class Product {

    @GeneratedValue
    @Id
    @Column(name="id", nullable = false)
    private long id;
    @Version
    @Column(name="version", nullable = false)
    private long version;

    @Basic
    @Column(name="name")
    private String name;

    @Basic
    @Column(name="price")
    private double price;

    @Basic
    @Column(name="quantity")
    private int quantity;

    @Basic
    @Column(name="barCode")
    private String barCode;

    @Basic
    @Column(name="uri")
    private String URI;

    @Basic
    @Column(name="description")
    private String description;

    @OneToMany(mappedBy = "product", cascade = CascadeType.MERGE, fetch = FetchType.LAZY )
    @JsonIgnore
    @ToString.Exclude
    private Collection<OrderDetails> orderDetails;

}

File: frontend/lib/models/Product.dart


class Product {
  final int id;
  final String name;
  final double price;
  final String description;
  int quantity;
  final String barCode;
  final String uri;
  String size;

   Product({
     required this.id,
    required this.name,
    required this.price,
     required this.description,
    this.quantity=1,
    required this.barCode,
    required this.uri,
     this.size="S",
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      price: json['price'],
      description: json['description'],
      quantity: json['quantity'],
      barCode: json['barCode'],
      uri: json['uri'],
      size: json['size']
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'price':price,
      'description' : description,
      'quantity':quantity,
      'barCode': barCode,
      'uri':uri,
      'size':size
    };
  }
}
